crates/fluvio/src/producer/accumulator.rs

```diff
@@ -218,6 +218,20 @@ impl RecordAccumulator {
     }
 }
 
+#[derive(Debug)]
+pub struct ProduceCompletionBatchEvent {
+    pub topic: String,
+    pub bytes_size: u64,
+    pub records_len: u64,
+    pub partition: PartitionId,
+    pub created_at: Instant,
+}
+
+pub type SharedProducerCallback = Arc<dyn ProducerCallback + Send + Sync>;
+pub trait ProducerCallback {
+    fn finished(&self, item: ProduceCompletionBatchEvent) -> BoxFuture<'_, anyhow::Result<()>>;
```

Comment:
does event can be reference rather than owned?

can this be simplified like here?
```
async fn finished(&self, item: &ProduceCompletionBatchEvent) -> Result<()>
```

or

Reply:
I guess not, it will require lifetimes when passing to channels